Docker/StorageApi/Writer.php

```diff
@@ -457,14 +467,55 @@ protected function uploadTable($source, $config = array())
                 $this->client->createTableAsync($bucketId, $tableName, $headerCsvFile, $options);
                 $options["columns"] = $config["columns"];
                 $options["withoutHeaders"] = true;
-                $this->client->writeTableAsync($config["destination"], $csvFile, $options);
+                if (is_dir($source)) {
+                    $options["delimiter"] = $config["delimiter"];
+                    $options["enclosure"] = $config["enclosure"];
+                    $this->writeSlicedTable($source, $config["destination"], $options);
+                } else {
+                    $csvFile = new CsvFile($source, $config["delimiter"], $config["enclosure"]);
+                    $this->client->writeTableAsync($config["destination"], $csvFile, $options);
+                }
             } else {
+                $csvFile = new CsvFile($source, $config["delimiter"], $config["enclosure"]);
                 $this->client->createTableAsync($bucketId, $tableName, $csvFile, $options);
             }
         }
     }
 
     /**
+     *
+     * Uploads a sliced table to storage api. Takes all files from the $source folder
+     *
+     * @param $source Slices folder
+     * @param $destination Destination table
+     * @param $options WriteTable options
+     */
+    protected function writeSlicedTable($source, $destination, $options)
+    {
+        $finder = new Finder();
+        $slices = $finder->files()->in($source);
+        $sliceFiles = [];
+        foreach ($slices as $slice) {
+            $sliceFiles[] = $slice->getPathname();
+        }
+        if (count($sliceFiles) === 0) {
+            return;
```

Comment:
nemelo by to spis hodit vyjimku? - to znamena, ze je tam prazdnej adresar a ten tam nema co delaat